File: Java/PizzaFactory.java
package teste;

public class PizzaFactory implements ComidaFactory{

	@Override
	public Comida criarComida(int id) {
		if(id == 0) {
			// Pizza mista
			Pizza p = new Pizza();
			p.nome = "Pizza mista";
			p.bordaComRecheio = false;
			p.sabor1 = "Queijo!";
			p.sabor2 = "Presunto!";
			p.comPalmito = false;
			return p;
		}else if(id == 1) {
			// Pizza portuguesa
			Pizza p = new Pizza();
			p.nome ="Pizza portuguesa";
			p.bordaComRecheio = true;
			p.sabor1 = "Ovo";
			p.sabor2 = "Presunto";
			p.comPalmito = true;
			return p;
		}else if(id == 2) {
			// Pizza de calabresa
			Pizza p = new Pizza();
			p.nome ="Pizza de calabresa com frango";
			p.bordaComRecheio = true;
			p.sabor1 = "Calabresa";
			p.sabor2 = "Frango";
			p.comPalmito = false;
			return p;
		}else {
			// Pizza mista
			Pizza p = new Pizza();
			p.nome = "Pizza mista";
			p.bordaComRecheio = false;
			p.sabor1 = "Queijo!";
			p.sabor2 = "Presunto!";
			p.comPalmito = false;
			return p;
		}
	}

}

File: PHP/PizzaFactory.php
<?php

require_once "./ComidaFactory.php";
require_once "./Pizza.php";

class PizzaFactory implements ComidaFactory{
    public function criarComida($id) : Comida{
        if($id == 0){
            // Pizza de calabresa
            $p = new Pizza();
            $p->nome = "Pizza de calabresa com frango!";
            $p->sabor1 = "Calabresa!";
            $p->sabor2 = "Frango";
            $p->comPalmito = false;
            $p->bordaComRecheio = true;
            return $p;
        }else if($id == 1){
            // Pizza de frango
            $p = new Pizza();
            $p->nome = "Pizza de frango";
            $p->sabor1 = "Frango!";
            $p->sabor2 = "Queijo";
            $p->comPalmito = true;
            $p->bordaComRecheio = true;
            return $p;
        }else{
            // Pizza mista
            $p = new Pizza();
            $p->nome = "Pizza mista";
            $p->sabor1 = "Presunto";
            $p->sabor2 = "Queijo";
            $p->comPalmito = false;
            $p->bordaComRecheio = true;
            return $p;
        }
    }
}